src/main/java/com/homework/lesson14/factorymethod/Factory.java

```diff
@@ -0,0 +1,15 @@
+package com.homework.lesson14.factorymethod;
+
+public class Factory {
+    public OS getCurrentOS(String inputos) {
+        OS os = null;
+        if (inputos.equals("windows")) {
```

Comment:
1) Why not use a switch case instead of if else?
2)  inputos could be null. So, you could get NPE.  Please fix it.